librz/cons/histogram.c

```diff
@@ -207,3 +210,204 @@ RZ_API RZ_OWN RzStrBuf *rz_histogram_vertical(RZ_NONNULL RzHistogramOptions *opt
+	rz_histogram_options_free(hist->opts);
+	free(hist);
+}
+
+RZ_API void rz_histogram_interactive_zoom_in(RzHistogramInteractive *hist) {
+	hist->zoom += ZOOM_DEFAULT;
+	if (hist->zoom > hist->size / hist->w + 31 - __builtin_clz(hist->w)) {
+		hist->zoom -= ZOOM_DEFAULT;
+	}
+}
+
+RZ_API void rz_histogram_interactive_zoom_out(RzHistogramInteractive *hist) {
+	hist->zoom -= ZOOM_DEFAULT;
+	if (hist->zoom == 0) {
+		hist->zoom = ZOOM_DEFAULT;
+	}
+}
+
+/**
+ * \brief Create the string buffer with the horisontal histogram
+ *
+ *		 █    ██      █             █                                        █
+ *		 █    ██      █             █                                       ██
+ *	 █   █    ██      █             █ █                                     ██
+ *	 ██  █    ██  █ ███             █ █              █                      ██   █
+ *	 ██  ██   ███ █ ███             █ █              █ █     █    █   █ █   ██   █
+ *	 ██  ██   ███ █ ███          █  █ █         █    █ █    ██    █   ███   ██   █
+ *	 ██  ██   ███ █ ███          █  █ █  █      █    █ █    ██    █   ███  ███ █ █
+ *	███  ██   ███ █ ███          █  █ █  █      █    █ █    ██    █   ███  █████ █
+ *	███████   ███ █ ███          █  █ █  █      █    ███    ██    █   ███  █████ █
+ *	███████   ███ █ ███   ██    ██  █ █  █  █   █    ███   ███   ██   ███  █████ █
+ *	███████  ████ █ ███   ██    ██  █ █  █  █   █    ███   ███   ██   ████ █████ █
+ * 	███████  ████ █ ███   ██    ██  █ █  ██ █   ██   ███   ███   ██   ████ █████ █
+ *	███████__████_█_███__███__█_██_████__████___██__████___███___██__█████_█████_█
+ *
+ * \param hist Information about the interactive histogram
+ * \param data A buffer with the numerical data in the format of one byte per value
+ * \param width Width of the histogram
+ * \param height Height of the histogram
+ */
+RZ_API RZ_OWN RzStrBuf *rz_histogram_interactive_horizontal(RZ_NONNULL RzHistogramInteractive *hist, const unsigned char *data) {
+	rz_return_val_if_fail(data, NULL);
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!buf) {
+		return NULL;
+	}
+
+	RzHistogramOptions *opts = hist->opts;
+	size_t i, j;
+	unsigned int width = hist->w;
+	unsigned int height = hist->h;
+	ut32 rows = height > 0 ? height : 10;
+	rows--;
+	const char *vline = opts->unicode ? RUNE_LINE_VERT : "|";
+	const char *block = opts->unicode ? UTF_BLOCK : "#";
+	const char *kol[5];
+	kol[0] = opts->pal->call;
+	kol[1] = opts->pal->jmp;
+	kol[2] = opts->pal->cjmp;
+	kol[3] = opts->pal->mov;
+	kol[4] = opts->pal->nop;
+	int zoom = hist->zoom;
+	int histogramwidth = hist->size;
+	int sizeofonebar = 1;
+	if (zoom > (histogramwidth / width)) {
+		sizeofonebar = zoom - (histogramwidth / width);
+	}
+	width /= sizeofonebar;
+	if (opts->color) {
+		int adder = 0;
+		if (sizeofonebar > 1) {
+			adder = hist->barnumber + 1 - width / 2;
+		} else {
+			adder = hist->barnumber + 1 - histogramwidth / (zoom * 2);
+		}
+		for (i = 0; i < rows; i++) {
+			size_t threshold = i * (0xff / rows);
```

Comment:
Please add more comments for the important pieces of code, e.g. what exactly calculated below.